jest/src/test/java/io/searchbox/core/search/aggregation/SignificantTermsAggregationIntegrationTest.java

```diff
@@ -0,0 +1,152 @@
+package io.searchbox.core.search.aggregation;
+
+import io.searchbox.common.AbstractIntegrationTest;
+import io.searchbox.core.Search;
+import io.searchbox.core.SearchResult;
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest;
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingResponse;
+import org.elasticsearch.test.ElasticsearchIntegrationTest;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author cfstout
+ */
+@ElasticsearchIntegrationTest.ClusterScope (scope = ElasticsearchIntegrationTest.Scope.SUITE, numDataNodes = 1)
```

Comment:
Ah ok then I know the problem: custom mapping is not propagated to (all) data nodes instantly but instead eventually... which sometimes is done after the test assertions are executed! You can use `waitForConcreteMappingsOnAll` to wait for the propagation to be completed but as you noted in `GetMappingIntegrationTest` even that doesn't work sometimes.